src/app/shared/models/notification-handler.ts

```diff
@@ -0,0 +1,24 @@
+export enum NotificationType {
+  SOON,
+  NOW,
+  REMINDER,
+  MISSED_SOON,
+  MISSED,
+  TEST
+}
+
+export interface SingleNotification {
+  task?: NotificationTaskInfo
```

Comment:
We can map these to task using the unique task identifier `index`.

Reply:
Also add a unique identifier for notifications for easy reference from the tasks model